lib/membrane_ffmpeg_swscale/scaler/scaler.ex

```diff
@@ -91,4 +98,21 @@ defmodule Membrane.FFmpeg.SWScale.Scaler do
         raise inspect(reason)
     end
   end
+
+  defp calculate_output_dims(%{output_width: nil, output_height: nil}, _original_dim),
+    do: raise("At least one dimension should be provided")
+
+  defp calculate_output_dims(%{output_width: nil} = state, {width, height}) do
+    output_width = div(state.output_height * width, height)
+    output_width = output_width + rem(output_width, 4)
```

Comment:
I've got some doubts concerning the `...+rem(output_width, 4)` part. What is the purpose of that? The adjustment of the resolution to the divisibility requirements is
is already covered by: https://github.com/membraneframework/membrane_ffmpeg_swscale_plugin/blob/e87dfea8994a79fa67e68b47a897e6ebae1e48f8/c_src/membrane_ffmpeg_swscale_plugin/scaler/scaler.c#L18

Reply:
@varsill yes I saw that this adjustment is covered by the `c function`, I just thought that this way the scaler doesn't need to add black paddings. Do you think it's better to leave this to the `scaler` ?

Reply:
Hi @gBillal !
Could you explain to me why adding `rem(output_width, 4)` will prevent the black paddings from being added? I've tried to prove that, but I didn't manage :/ 
I understand that subtracting of `rem(outuput_width, 4)` would make our the `output_width` divisible by 4 (which is required by the YUV format and which is provided in the C function), but when it comes to adding, I don 't see the relation.

Reply:
@varsill sorry, you're right. Looks like I forget my maths, it should be a subtraction and not an addition.
Will fix this.

Reply:
No problem! Now I understand ;) 
But to be honest I would simply leave it as `output_width = div(state.output_height *width, height)` (and the corresponding operation for `output_height` without the subtraction, and let the function in C do the job - the function in C still might need to add paddings even if the particular dimension value is divisible by 4.